### PR TITLE
expression: let `PushDownNot` does not change the argument

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/10344

### What is changed and how it works?

`PushDownNot` directly change its argument `expr`, and if the `expr` is referenced in other places, it will cause wrong results. This PR keeps the argument unchanged.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/logical_plan_test.go [hunk 1]
```diff
@@ -395,7 +395,7 @@ func (s *testPlanSuite) TestSimplifyOuterJoin(c *C) {
 		},
 		{
 			sql:      "select * from t t1 left join t t2 on t1.b = t2.b where not (t1.c > 1 and t2.c > 1);",
-			best:     "Join{DataScan(t1)->DataScan(t2)}(test.t1.b,test.t2.b)->Sel([not(and(le(test.t1.c, 1), le(test.t2.c, 1)))])->Projection",
```
Comment (winoros) [inline planner/core/logical_plan_test.go]: So this one is wrong previously? :joy:

Comment (lamxTyler) [reply planner/core/logical_plan_test.go]: Yes...

File: expression/util.go [hunk 2]
```diff
@@ -289,6 +289,14 @@ var symmetricOp = map[opcode.Op]opcode.Op{
 	opcode.NullEQ: opcode.NullEQ,
 }
 
+func pushDownNot(ctx sessionctx.Context, exprs []Expression, not bool) []Expression {
```
Comment (XuHuaiyu) [inline expression/util.go]: doPushDownNot ?

**Pull request merged**